### DISCUSSION TITLE
Stencil private Jest typings overriding standard jest-dom types in Vite environment

### DISCUSSION THREAD

**Discussion by AlmeroSteyn** [stenciljs/core#6060] [Stencil Testing]
I'm using the @arcgis libraries in a project and Stencil is installed as a dependency from those libraries. Once Stencil was updated from 4.18 to 4.20 the jest types inside @stencil/core/internal/utils/stencil-private.d.ts started overriding the toHaveAttribute function definition I need from @testing-library/jest-dom

Should this private types file be published to the npm package?

The only way I can fix this is to create another types file to again override the global jest definition for this function.

Any other ideas?

**Comment by christian-bromann** [comment]
> Any other ideas?

Not at the moment. I would recommend though to may try a different setup with WebdriverIO or Playwright. These tools do test the component in an actual browser instead of using a fake implementation (jest-dom).

**Reply by AlmeroSteyn** [reply]
Thank you for the reply and suggestions. 

Please note that jest-dom is not a DOM implementation but a very popular plugin library of custom DOM test matchers created by the @testing-library team. 

The way that Stencil has extended the matchers in JEST in a file marked as **private** breaks this very handy library in TypeScript.

Could not doing this anymore (or not exposing it in the npm package) perhaps be a more efficient suggestion than entire projects needing to hack around this or recode their entire unit test pack?

**Comment by wa1id** [comment]
Bumped into the same issue. Overriding with my own types file is not so clean. Have you come up with a better solution @AlmeroSteyn ?

**Reply by AlmeroSteyn** [reply]
No, still have to override the types to have it work.
